cpu/esp8266/Makefile.include

```diff
@@ -0,0 +1,175 @@
+# check some environment variables first
+ifndef NEWLIB_DIR
+    $(info NEWLIB_DIR should be defined as /path/to/newlib directory)
+    $(info NEWLIB_DIR is set by default to /opt/esp/newlib-xtensa)
+    export NEWLIB_DIR=/opt/esp/newlib-xtensa
+endif
+
+ifndef SDK_DIR
+    $(info SDK_DIR should be defined as /path/to/sdk directory)
+    $(info SDK_DIR is set by default to /opt/esp/esp-open-sdk/sdk)
+    export SDK_DIR=/opt/esp/esp-open-sdk/sdk
+endif
+
+export CPU ?= esp8266
+export TARGET_ARCH ?= xtensa-lx106-elf
+
+INCLUDES += -I$(NEWLIB_DIR)/$(TARGET_ARCH)/include
+INCLUDES += -I$(RIOTBOARD)/common/$(CPU)/include
+INCLUDES += -I$(RIOTCPU)/$(CPU)
+INCLUDES += -I$(RIOTCPU)/$(CPU)/espressif
+
+CFLAGS  += -DRIOT_OS -DESP_OPEN_SDK
+CFLAGS  += -Wno-unused-parameter -Wformat=0
+CFLAGS  += -mlongcalls -mtext-section-literals
+ASFLAGS += --longcalls --text-section-literals
+
+NETIF_NUMOF=0
+
+ifeq ($(SDK), 1)
+    INCLUDES += -I$(SDK_DIR)/include
+    CFLAGS   += -DSDK_USED -DUSE_US_TIMER
+    ifeq ($(SDK_INT_HANDLING), 1)
+        CFLAGS += -DSDK_INT_HANDLING
+    endif
+else
+    INCLUDES += -I$(SDK_DIR)/include
+    CFLAGS   += -DSDK_NOT_USED
+endif
+
+CFLAGS_OPT = -fzero-initialized-in-bss -O2
```

Comment:
Almost all platforms default to ```-Os```. Any specific reason for O2? Does that have a large impact?

Reply:
@kaspar030 According to gcc man page, the default optimization level is ```-O0```. Commands 
```
xtensa-lx106-elf-gcc -Q --help=optimizers
xtensa-lx106-elf-gcc -Q --help=optimizers -O0
```
give the same results. That is, the default optimization level of xtensa version of gcc seems to be ```-O0```. I would like to leave the option ``` -O2``` for the following reasons:

- I tested all the code with this optimization.
- Most of other ESP8266 projects I saw use this optimization option
- I2C is implemented as bit-banging protocol in software. The delays used there are measured with this optimization option. Other optimizations could result in wrong timings

Reply:
The code space is usually not the problem for ESP8266 modules. They have at least 1 MByte flash size. The problem is rather the Instruction RAM (IRAM) which has only 32 kBytes. Certain parts of code have to reside in IRAM, e.g., ISRs.

Reply:
ok!